src/provider.tsx

```diff
@@ -62,8 +62,16 @@ class LDProvider extends React.Component<ProviderConfig, HocState> implements En
 
   initLDClient = async () => {
     const { clientSideID, flags, options, user } = this.props;
+    let ldClient = await this.props.ldClient;
```

Comment:
Does this work? I don't think await will make a difference when destructuring/assigning.

```suggestion
    const { clientSideID, flags, options, user, ldClient } = this.props;
```

Reply:
True, it won't make a difference when destructuring, but would still need a line to await and (re)assign the `ldClient` to the result before going on to use it (and fiddling around with `const` or variable names to allow the assignment), so I don't think this would add much in terms of readability.